test/test_mahalanobis_mixin.py

```diff
@@ -150,13 +150,9 @@ def check_is_distance_matrix(pairwise):
   assert np.array_equal(pairwise, pairwise.T)  # symmetry
   assert (pairwise.diagonal() == 0).all()  # identity
   # triangular inequality
-  rows, cols = pairwise.shape
-  tol = np.finfo(float).eps
-  for i in range(rows):
-    for j in range(i, cols):
-      direct_dist = pairwise[i, j]
-      triangle_dist = np.add.outer(pairwise[i], pairwise[:, j])
-      assert (direct_dist <= triangle_dist + tol).all()
+  tol = 1e-15
+  assert (pairwise <= pairwise[:, :, np.newaxis]
+          + pairwise[:, np.newaxis, :] + tol).all()
```

Comment:
I think this would work, checking that each `pairwise[i, j] <= pairwise[i, :] + pairwise[:, j]` (the strict version of this would be with `pairwise.T[:, :, np.newaxis] + pairwise[:, np.newaxis, :]` but since pairwise is already checked to be symmetric we can avoid the transpose)
For the value `1e-15`, the test failed with 1e-16 for lsml, when sometimes running the tests, and sometimes not. I think 1e-15 is a reasonable value to set for now, but I will also make a commit to set all random_states (this made me realized they were not)